spec/models/upload_json_spec.rb

```diff
@@ -58,8 +58,16 @@
     }.to_json
 
     upload_json = build(:upload_json, competition_id: competition.id, results_json_str: results_json)
+    temporary_results_data = upload_json.temporary_results_data
+    CompetitionResultsImport.import_temporary_results(
+      competition,
+      temporary_results_data,
+      mark_result_submitted: false,
+      store_uploaded_json: true,
+      results_json_str: upload_json.results_json_str,
+    )
 
-    expect(upload_json.import_to_inbox).to be true
+    expect(upload_json).to be_valid
```

Comment:
You should move this test before the actual import. Trying to import from a (potentially) invalid JSON doesn't make any sense.